link standalone content (#934)
hashicorp/design-system: website/docs/components/link/standalone/partials/guidelines/guidelines.md

## Usage

### When to use

- To navigate to a new destination, either internally or externally.
- As a standalone element or within a Button Set, use a **(Link) Standalone**

### When not to use

- Within a body of text, use a **Link (Inline)**
- To trigger an action or event, like a form submission, use a [Button](/components/button/overview)

### Size

<Hds::Link::Standalone @size="large" @text="Large" @icon="arrow-right" @iconPosition="trailing" @route="components" />

<Hds::Link::Standalone @size="medium" @text="Medium" @icon="arrow-right" @iconPosition="trailing" @route="components" />

<Hds::Link::Standalone @size="small" @text="Small" @icon="arrow-right" @iconPosition="trailing" @route="components" />

> Medium is the preferred size, but use a link size that best fits the UI.

Reviewer: ```suggestion
In most cases use a medium size, but use a link size that best fits the UI.
```

  <Hds::Link::Standalone @text="View billing" @icon="arrow-right" @iconPosition="trailing" @color="secondary" @route="components" />
  <Hds::Link::Standalone @text="Submit feedback" @icon="external-link" @iconPosition="trailing" @color="secondary" @route="components" />
  <Hds::Link::Standalone @text="Consul documentation" @icon="docs-link" @iconPosition="trailing" @color="secondary" @route="components" />
  <Hds::Link::Standalone @text="Service mesh tutorial" @icon="learn-link" @iconPosition="trailing" @color="secondary" @route="components" />

!!! Info

If adding a specific icon is relevant to the user experience, and none of the icons available in Flight fit your use case, consider [requesting a custom icon](https://github.com/hashicorp/design-system/issues/new/choose).
!!!

## Common patterns

### Within button sets

**Link (Standalone)** can commonly be seen within **button sets**.

<Hds::ButtonSet>
  <Hds::Link::Standalone @text="Previous" @icon="chevron-left" @iconPosition="leading" @color="primary" @route="components" />
  <Hds::Button @text="Submit" @color="primary" />
  <Hds::Button @text="Cancel" @color="secondary" />
</Hds::ButtonSet>

> Often when included in a ButtonSet, Link (Standalone) is used for navigating between steps in a multi-page form (ie. directing the user back to the previous step). As such, they should be displayed as the left most button in the row.

Reviewer: ```suggestion
When included in a ButtonSet, Link (Standalone) is generally used for navigating between steps in a multi-page form (ie. directing the user back to the previous step) or for linking to external documentation. As such, they should be displayed as the right most button in the row.
```

This corresponds with the button alignment/ordering/grouping research.

## Usage

### When to use

- To navigate to a new destination, either internally or externally.
- As a standalone element or within a Button Set, use a **(Link) Standalone**

### When not to use

- Within a body of text, use a **Link (Inline)**
- To trigger an action or event, like a form submission, use a [Button](/components/button/overview)

### Size

<Hds::Link::Standalone @size="large" @text="Large" @icon="arrow-right" @iconPosition="trailing" @route="components" />

> <Hds::Link::Standalone @size="medium" @text="Medium" @icon="arrow-right" @iconPosition="trailing" @route="components" />

Reviewer 2: ```suggestion

<Hds::Link::Standalone @size="medium" @text="Medium" @icon="arrow-right" @iconPosition="trailing" @route="components" />

```
These need some breathing room, visually.

Author: Agreed, but I kinda punted on that for this first pass. I'll commit this to see if it helps.

<Hds::Link::Standalone @size="small" @text="Small" @icon="arrow-right" @iconPosition="trailing" @route="components" />

Medium is the preferred size, but use a link size that best fits the UI.

For example:

- Don't use large links in tables.

### Type

#### Primary

Use the `primary` variant as the default and for more important links. 

For example when navigating the user to `View clusters`:

<Hds::Link::Standalone @text="View clusters" @icon="arrow-right" @iconPosition="trailing" @type="primary" @route="components" />

!!! Info

Take care when needing multiple links on a page; using too many primary links can detract from the end goal.
!!!
#### Secondary

Use the Secondary variant for less important links, when the action color can't be used, or when creating a list of links.

For example in an Install page, 'Learn more about Vault'

<Hds::Link::Standalone @text="View clusters" @icon="external-link" @iconPosition="trailing" @color="secondary" @route="components" />

### State
#### Primary

> TBD Image here?

Author: @jorytindall  this is an example where we have this image in the existing guidelines but i'm not sure it is possible to recreate in code

Author: I think either way I should move these to the Specifications tab

Author: moved these to the correct place and added images
